browser/youtube_script_injector/android_youtube_script_injector_browsertest.cc

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "base/path_service.h"
+#include "brave/components/constants/brave_paths.h"
+#include "chrome/test/base/android/android_browser_test.h"
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "content/public/common/content_client.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+
+namespace {
+constexpr char kPipScriptInjected[] = "window._pipScriptInjected === true";
+}  // namespace
+
+class AndroidYouTubeScriptInjectorBrowserTest : public PlatformBrowserTest {
+ public:
+  AndroidYouTubeScriptInjectorBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  ~AndroidYouTubeScriptInjectorBrowserTest() override = default;
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    PlatformBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void SetUpOnMainThread() override {
+    PlatformBrowserTest::SetUpOnMainThread();
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+
+    base::FilePath test_data_dir = GetTestDataDir();
+
+    https_server_.ServeFilesFromDirectory(test_data_dir);
+    content::SetupCrossSiteRedirector(&https_server_);
+    ASSERT_TRUE(https_server_.Start());
+  }
+
+  base::FilePath GetTestDataDir() {
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    return base::PathService::CheckedGet(brave::DIR_TEST_DATA);
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    PlatformBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* web_contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+ protected:
+  // Must use HTTPS because `youtube.com` is in Chromium's HSTS preload list.
+  net::EmbeddedTestServer https_server_;
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// TESTS
+
+IN_PROC_BROWSER_TEST_F(AndroidYouTubeScriptInjectorBrowserTest,
```

Comment:
in addition to the tests to check that the property overrides work correctly, this is also missing the same page navigation test that we discussed.